core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -342,6 +320,87 @@ class ExecutorSuite extends SparkFunSuite
     }
   }
 
+  test("Send task executor metrics in DirectTaskResult") {
+    // Run a successful, trivial result task
+    // We need to ensure, however, that executor metrics are polled after the task is started
+    // so this requires some coordination using ExecutorSuiteHelper.
+    val conf = new SparkConf().setMaster("local").setAppName("executor suite test")
+    sc = new SparkContext(conf)
+    val serializer = SparkEnv.get.closureSerializer.newInstance()
+    ExecutorSuiteHelper.latches = new ExecutorSuiteHelper
+    val resultFunc =
+      (context: TaskContext, itr: Iterator[Int]) => {
+        ExecutorSuiteHelper.latches.latch1.await(300, TimeUnit.MILLISECONDS)
+        ExecutorSuiteHelper.latches.latch2.countDown()
+        ExecutorSuiteHelper.latches.latch3.await(500, TimeUnit.MILLISECONDS)
```

Comment:
would help to have a short comment about the purpose of these latches.  Eg. "latch2 tells the metricsPoller to poll and send results, and latch3 waits till the poller has sent its results"

(I don't see the point of latch1, so you'd probably switch to just using latch1 & latch2)

Reply:
I can't recall what I was thinking back then. ;-) Maybe that ExecutorSuiteHelper has 3 latches so I should use all 3?
You're right, I only need 2 latches.
I'll add a comment about their use.